### DISCUSSION TITLE
Is there a way to write macros in multiple lines?

### DISCUSSION THREAD

**Discussion by fespinoza** [krzysztofzablocki/SourceryPro-Feedback#21] [Q&A] [answered]
I have a macro that looks like

```
{% macro defaultValue propertyName propertyTypeName %}{% typed propertyName as String %}{% typed propertyTypeName as TypeName %}{% if propertyTypeName.name == "String" %}""{% else %}{% if propertyTypeName.name == "Int" or propertyTypeName.name == "CGFloat" or propertyTypeName.name == "Float" or propertyTypeName.name == "Double" %}0{% else %}<#defaultValue-{{propertyName}}#>{% endif %}{% endif %}{% endmacro %}
```

which ideally I would like to write in multiple lines

```
{% macro defaultValue propertyName propertyTypeName %}
{% typed propertyName as String %}
{% typed propertyTypeName as TypeName %}
{% if propertyTypeName.name == "String" %}
""
{% else %}
{% if propertyTypeName.name == "Int" or propertyTypeName.name == "CGFloat" or propertyTypeName.name == "Float" or propertyTypeName.name == "Double" %}0{% else %}
<#defaultValue-{{propertyName}}#>
{% endif %}
{% endif %}
{% endmacro %}
```

The problem Is that those new lines are shown in the output.

I'm not sure if there is a way to achieve this. Is it a stencil issue? sourcery pro missing feature? etc.

**Comment by fespinoza** [comment]
P.S.: the link for "sorcery documentation" in the Mac app is "broken", as in renders raw html
<img width="406" alt="Screen Shot 2022-03-27 at 22 29 51" src="https://user-images.githubusercontent.com/167989/160299705-76200d13-1794-44c2-a528-9de0ef10f6ff.png">

**Comment by krzysztofzablocki** [comment, marked as answer]
generally a stencil issue, but if you put the macros at the top of the file and enable trim whitespace in Pro you should get rid of any additional spaces before normal codegen code

**Reply by fespinoza** [reply]
Thanks for the answer, but that doesn't really work

<img width="1650" alt="Screen Shot 2022-03-28 at 10 21 32" src="https://user-images.githubusercontent.com/167989/160356633-1119a489-25a3-4490-bb19-e4174120872e.png">

In this case the problem is the new lines added by having the macro as a multiple line definition.
but I get it, it's an stencil issue
